File: laboratoire 2/CUnitConvert/practica02.c
#include <stdio.h>
#include "libreria_energia.h"

float menu_seleccion(void);
float menu_conversionJoule(void);
float menu_conversionCaloria(void);
float menu_conversionKCaloria(void);
float menu_conversionBTU(void);
float menu_conversionKWHR(void);
float menu_conversionHPHR(void);
float menu_conversionPielb(void);
float menu_conversionLitroAtm(void);

float resultado, numero;
int opcion;

float main(void)
{
    menu_seleccion(); 
}

float menu_seleccion(void)
{
    
    printf("\nEscriba la cantidad, después el número correspondiente a la unidad que quiere convertir");
    
    printf("\n1 -- Joule");
    printf("\n2 -- Caloría");
    printf("\n3 -- KCaloría");
    printf("\n4 -- BTU - British Thermal Unit");
    printf("\n5 -- KW-HR");
    printf("\n6 -- HP-HR");
    printf("\n7 -- Pie-Libra");
    printf("\n8 -- Litro-Atmosfera\n");  
    scanf("%f",&numero);
    scanf("%d",&opcion);
    printf("\n%f",numero);
    
    switch(opcion)
    {
        case 1:
        menu_conversionJoule();
        break;
        
        case 2:
        menu_conversionCaloria();
        break;
        
        case 3:
        menu_conversionKCaloria();
        break;
        
        case 4:
        menu_conversionBTU();
        break;
        
        case 5:
        menu_conversionKWHR();
        break;
        
        case 6:
        menu_conversionHPHR();
        break;
        
        case 7:
        menu_conversionPielb();
        break;
        
        case 8:
        menu_conversionLitroAtm();
        break;
    }
        
    printf ("El resultado es %f\n", resultado);
}

float menu_conversionJoule(void)
{
    
    printf("\n\nElija la unidad a la que desea convertir");
    
    printf("\n1 -- Caloría");
    printf("\n2 -- KCaloría");
    printf("\n3 -- BTU - British Thermal Unit");
    printf("\n4 -- KW-HR");
    printf("\n5 -- HP-HR");
    printf("\n6 -- Pie-Libra");
    printf("\n7 -- Litro-Atmosfera\n");
    
    scanf("%d",&opcion);
    
    switch(opcion)
    {
        case 1:
        conversionJouleCaloria(numero);
        break;
        
        case 2:
        conversionJouleKCaloria(numero);
        break;
        
        case 3:
        conversionJouleBTU(numero);
        break;
        
        case 4:
        conversionJouleKWHR(numero);
        break;
        
        case 5:
        conversionJouleHPHR(numero);
        break;
        
        case 6:
        conversionJoulePielb(numero);
        break;
        
        case 7:
        conversionJouleLitroAtm(numero);
        break;
        
        case 8:
        
        break;
    }
}

float menu_conversionCaloria(void)
{
    printf("Elija la unidad a la que desea convertir");
    printf("\n1 -- Joule");
    printf("\n2 -- KCaloría");
    printf("\n3 -- BTU - British Thermal Unit");
    printf("\n4 -- KW-HR");
    printf("\n5 -- HP-HR");
    printf("\n6 -- Pie-Libra");
    printf("\n7 -- Litro-Atmosfera\n");
    
    scanf("%d",&opcion);
    
    switch(opcion)
    {
        case 1:
        conversionCaloriaJoule(numero);
        break;
        
        case 2:
        conversionCaloriaKCaloria(numero);
        break;
        
        case 3:
        conversionCaloriaBTU(numero);
        break;
        
        case 4:
        conversionCaloriaKWHR(numero);
        break;
        
        case 5:
        conversionCaloriaHPHR(numero);
        break;
        
        case 6:
        conversionCaloriaPielb(numero);
        break;
        
        case 7:
        conversionCaloriaLitroAtm(numero);
        break;
        
        case 8:
        
        break;
    }
    return 0;
}

float menu_conversionKCaloria(void)
{
    printf("\n\nElija la unidad a la que desea convertir");
    printf("\n1 -- Joule");
    printf("\n2 -- Caloría");
    printf("\n3 -- BTU - British Thermal Unit");
    printf("\n4 -- KW-HR");
    printf("\n5 -- HP-HR");
    printf("\n6 -- Pie-Libra");
    printf("\n7 -- Litro-Atmosfera\n");
    
    scanf("%d",&opcion);
    
    switch(opcion)
    {
        case 1:
        conversionKCaloriaJoule(numero);
        break;
        
        case 2:
        conversionKCaloriaCaloria(numero);
        break;
        
        case 3:
        conversionKCaloriaBTU(numero);
        break;
        
        case 4:
        conversionKCaloriaKWHR(numero);
        break;
        
        case 5:
        conversionKCaloriaHPHR(numero);
        break;
        
        case 6:
        conversionKCaloriaPielb(numero);
        break;
        
        case 7:
        conversionKCaloriaLitroAtm(numero);
        break;
        
        case 8:
        
        break;
    }
    return 0;
}

float menu_conversionBTU(void)
{
    printf("\n\nElija la unidad a la que desea convertir");
    printf("\n1 -- Joule");
    printf("\n2 -- Caloría");
    printf("\n3 -- KCaloría");
    printf("\n4 -- KW-HR");
    printf("\n5 -- HP-HR");
    printf("\n6 -- Pie-Libra");
    printf("\n7 -- Litro-Atmosfera\n");
    
    scanf("%d",&opcion);
    
    switch(opcion)
    {
        case 1:
        conversionBTUJoule(numero);
        break;
        
        case 2:
        conversionBTUCaloria(numero);
        break;
        
        case 3:
        conversionBTUKCaloria(numero);
        break;
        
        case 4:
        conversionBTUKWHR(numero);
        break;
        
        case 5:
        conversionBTUHPHR(numero);
        break;
        
        case 6:
        conversionBTUPielb(numero);
        break;
        
        case 7:
        conversionBTULitroAtm(numero);
        break;
        
        case 8:
        
        break;
    }
    return 0;
}

float menu_conversionKWHR(void)
{
    printf("\n\nElija la unidad a la que desea convertir");
    printf("\n1 -- Joule");
    printf("\n2 -- Caloría");
    printf("\n3 -- KCaloría");
    printf("\n4 -- BTU - British Thermal Unit");
    printf("\n5 -- HP-HR");
    printf("\n6 -- Pie-Libra");
    printf("\n7 -- Litro-Atmosfera\n");
    
    scanf("%d",&opcion);
    
    switch(opcion)
    {
        case 1:
        conversionKWHRJoule(numero);
        break;
        
        case 2:
        conversionKWHRCaloria(numero);
        break;
        
        case 3:
        conversionKWHRKCaloria(numero);
        break;
        
        case 4:
        conversionKWHRBTU(numero);
        break;
        
        case 5:
        conversionKWHRHPHR(numero);
        break;
        
        case 6:
        conversionKWHRPielb(numero);
        break;
        
        case 7:
        conversionKWHRLitroAtm(numero);
        break;
        
        case 8:
        
        break;
    }
    return 0;
}

float menu_conversionHPHR(void)
{
    printf("\n\nElija la unidad a la que desea convertir");
    printf("\n1 -- Joule");
    printf("\n2 -- Caloría");
    printf("\n3 -- KCaloría");
    printf("\n4 -- BTU - British Thermal Unit");
    printf("\n5 -- KW-HR");
    printf("\n6 -- Pie-Libra");
    printf("\n7 -- Litro-Atmosfera\n");
    
    scanf("%d",&opcion);
    
    switch(opcion)
    {
        case 1:
        conversionHPHRJoule(numero);
        break;
        
        case 2:
        conversionHPHRCaloria(numero);
        break;
        
        case 3:
        conversionHPHRKCaloria(numero);
        break;
        
        case 4:
        conversionHPHRBTU(numero);
        break;
        
        case 5:
        conversionHPHRKWHR(numero);
        break;
        
        case 6:
        conversionHPHRPielb(numero);
        break;
        
        case 7:
        conversionHPHRLitroAtm(numero);
        break;
        
        case 8:
        
        break;
    }
    return 0;
}

float menu_conversionPielb(void)
{
    printf("\n\nElija la unidad a la que desea convertir");
    printf("\n1 -- Joule");
    printf("\n2 -- Caloría");
    printf("\n3 -- KCaloría");
    printf("\n4 -- BTU - British Thermal Unit");
    printf("\n5 -- KW-HR");
    printf("\n6 -- HP-HR");
    printf("\n7 -- Litro-Atmosfera\n");
    
    scanf("%d",&opcion);
    
    switch(opcion)
    {
        case 1:
        conversionPielbJoule(numero);
        break;
        
        case 2:
        conversionPielbCaloria(numero);
        break;
        
        case 3:
        conversionPielbKCaloria(numero);
        break;
        
        case 4:
        conversionPielbBTU(numero);
        break;
        
        case 5:
        conversionPielbKWHR(numero);
        break;
        
        case 6:
        conversionPielbHPHR(numero);
        break;
        
        case 7:
        conversionPielbLitroAtm(numero);
        break;
        
        case 8:
        
        break;
    }
    return 0;
}

float menu_conversionLitroAtm(void)
{
    printf("\n\nElija la unidad a la que desea convertir");
    printf("\n1 -- Joule");
    printf("\n2 -- Caloría");
    printf("\n3 -- KCaloría");
    printf("\n4 -- BTU - British Thermal Unit");
    printf("\n5 -- KW-HR");
    printf("\n6 -- HP-HR");
    printf("\n7 -- Pie-Libra\n");
    
    scanf("%d",&opcion);
    
    switch(opcion)
    {
        case 1:
        conversionLitroAtmJoule(numero);
        break;
        
        case 2:
        conversionLitroAtmCaloria(numero);
        break;
        
        case 3:
        conversionLitroAtmKCaloria(numero);
        break;
        
        case 4:
        conversionLitroAtmBTU(numero);
        break;
        
        case 5:
        conversionLitroAtmKWHR(numero);
        break;
        
        case 6:
        conversionLitroAtmHPHR(numero);
        break;
        
        case 7:
        conversionLitroAtmPielb(numero);
        break;
        
        case 8:
        
        break;
    }
}

File: laboratoire 2/PythonUnitConvert/practica02.py
#include <stdio.h>
#include "libreria_energia.h"

import libreria_energia
import objeto as o

resultado=0.00000001
numero=0.0000001
opcion=0

def main():
    menu_seleccion(); 


def menu_seleccion():
    
    print("\nEscriba la cantidad, el numero correspondiente a la unidad que quiere convertir")
    
    print("\n1 -- Joule")
    print("\n2 -- Caloría")
    print("\n3 -- KCaloría")
    print("\n4 -- BTU - British Thermal Unit")
    print("\n5 -- KW-HR")
    print("\n6 -- HP-HR")
    print("\n7 -- Pie-Libra")
    print("\n8 -- Litro-Atmosfera\n");  
    
    print(numero)
    opcion=int(input("option? "))
    print("\n%f",numero)
    switch_dict = {
            1:menu_conversionJoule,
            2:menu_conversionCaloria,
            3:menu_conversionKCaloria,
            4:menu_conversionBTU,
            5:menu_conversionKWHR,
            6:menu_conversionHPHR,
            7:menu_conversionPielb,
            8:menu_conversionLitroAtm,
            }
    resultado=switch_dict[opcion]()
        
    print("El resultado es " + str(resultado))
    return


def menu_conversionJoule():
    
    print("\n\nElija la unidad a la que desea convertir")
    
    print("\n1 -- Caloría")
    print("\n2 -- KCaloría")
    print("\n3 -- BTU - British Thermal Unit")
    print("\n4 -- KW-HR")
    print("\n5 -- HP-HR")
    print("\n6 -- Pie-Libra")
    print("\n7 -- Litro-Atmosfera\n")
    
    opcion=int(input("option? "))
    switch_dict = {
            1:o.conversionJouleCaloria,
            2:o.conversionJouleKCaloria,
            3:o.conversionJouleBTU,
            4:o.conversionJouleKWHR,
            5:o.conversionJouleHPHR,
            6:o.conversionJoulePielb,
            7:o.conversionJouleLitroAtm,
            }
    numero=int(input("numero? "))
    return switch_dict[opcion](numero)




def menu_conversionCaloria():
    print("Elija la unidad a la que desea convertir")
    print("\n1 -- Joule")
    print("\n2 -- KCaloría")
    print("\n3 -- BTU - British Thermal Unit")
    print("\n4 -- KW-HR")
    print("\n5 -- HP-HR")
    print("\n6 -- Pie-Libra")
    print("\n7 -- Litro-Atmosfera\n")
    
    opcion=int(input("option? "))
    numero=int(input("numero? "))
    switch_dict = {
            1:o.conversionCaloriaJoule,
            2:o.conversionCaloriaKCaloria,
            3:o.conversionCaloriaBTU,
            4:o.conversionCaloriaKWHR,
            5:o.conversionCaloriaHPHR,
            6:o.conversionCaloriaPielb,
            7:o.conversionCaloriaLitroAtm,
            }
    numero=int(input("numero? "))
    return switch_dict[opcion](numero)    



def menu_conversionKCaloria():
    print("\n\nElija la unidad a la que desea convertir")
    print("\n1 -- Joule")
    print("\n2 -- Caloría")
    print("\n3 -- BTU - British Thermal Unit")
    print("\n4 -- KW-HR")
    print("\n5 -- HP-HR")
    print("\n6 -- Pie-Libra")
    print("\n7 -- Litro-Atmosfera\n")
    
    opcion=int(input("option? "))

    switch_dict = {
            1:o.conversionKCaloriaJoule,
            2:o.conversionKCaloriaCaloria,
            3:o.conversionKCaloriaBTU,
            4:o.conversionKCaloriaKWHR,
            5:o.conversionKCaloriaHPHR,
            6:o.conversionKCaloriaPielb,
            7:o.conversionKCaloriaLitroAtm,
            }
    numero=int(input("numero? "))
    return switch_dict[opcion](numero) 



def menu_conversionBTU():
    print("\n\nElija la unidad a la que desea convertir")
    print("\n1 -- Joule")
    print("\n2 -- Caloría")
    print("\n3 -- KCaloría")
    print("\n4 -- KW-HR")
    print("\n5 -- HP-HR")
    print("\n6 -- Pie-Libra")
    print("\n7 -- Litro-Atmosfera\n")

    opcion=int(input("option? "))

    switch_dict = {
            1:o.conversionBTUJoule,
            2:o.conversionBTUCaloria,
            3:o.conversionBTUKCaloria,
            4:o.conversionBTUKWHR,
            5:o.conversionBTUHPHR,
            6:o.conversionBTUPielb,
            7:o.conversionBTULitroAtm,
            }
    numero=int(input("numero? "))
    return switch_dict[opcion](numero)     



def menu_conversionKWHR():
    print("\n\nElija la unidad a la que desea convertir")
    print("\n1 -- Joule")
    print("\n2 -- Caloría")
    print("\n3 -- KCaloría")
    print("\n4 -- BTU - British Thermal Unit")
    print("\n5 -- HP-HR")
    print("\n6 -- Pie-Libra")
    print("\n7 -- Litro-Atmosfera\n")

    opcion=int(input("option? "))

    switch_dict = {
            1:o.conversionKWHRJoule,
            2:o.conversionKWHRCaloria,
            3:o.conversionKWHRKCaloria,
            4:o.conversionKWHRBTU,
            5:o.conversionKWHRHPHR,
            6:o.conversionKWHRPielb,
            7:o.conversionKWHRLitroAtm,
            }
    numero=int(input("numero? "))
    return switch_dict[opcion](numero)  



def menu_conversionHPHR():
    print("\n\nElija la unidad a la que desea convertir")
    print("\n1 -- Joule")
    print("\n2 -- Caloría")
    print("\n3 -- KCaloría")
    print("\n4 -- BTU - British Thermal Unit")
    print("\n5 -- KW-HR")
    print("\n6 -- Pie-Libra")
    print("\n7 -- Litro-Atmosfera\n")
 
    opcion=int(input("option? "))

    switch_dict = {
            1:o.conversionHPHRJoule,
            2:o.conversionHPHRCaloria,
            3:o.conversionHPHRKCaloria,
            4:o.conversionHPHRBTU,
            5:o.conversionHPHRKWHR,
            6:o.conversionHPHRPielb,
            7:o.conversionHPHRLitroAtm,
            }
    numero=int(input("numero? "))
    return switch_dict[opcion](numero)  



def menu_conversionPielb():
    print("\n\nElija la unidad a la que desea convertir")
    print("\n1 -- Joule")
    print("\n2 -- Caloría")
    print("\n3 -- KCaloría")
    print("\n4 -- BTU - British Thermal Unit")
    print("\n5 -- KW-HR")
    print("\n6 -- HP-HR")
    print("\n7 -- Litro-Atmosfera\n")

    opcion=int(input("option? "))

    switch_dict = {
            1:o.conversionPielbJoule,
            2:o.conversionPielbCaloria,
            3:o.conversionPielbKCaloria,
            4:o.conversionPielbBTU,
            5:o.conversionPielbKWHR,
            6:o.conversionPielbHPHR,
            7:o.conversionPielbLitroAtm,
            }
    numero=int(input("numero? "))
    return switch_dict[opcion](numero)  


def menu_conversionLitroAtm():
    print("\n\nElija la unidad a la que desea convertir")
    print("\n1 -- Joule")
    print("\n2 -- Caloría")
    print("\n3 -- KCaloría")
    print("\n4 -- BTU - British Thermal Unit")
    print("\n5 -- KW-HR")
    print("\n6 -- HP-HR")
    print("\n7 -- Pie-Libra\n")


    opcion=int(input("option? "))

    switch_dict = {
            1:o.conversionLitroAtmJoule,
            2:o.conversionLitroAtmCaloria,
            3:o.conversionLitroAtmKCaloria,
            4:o.conversionLitroAtmBTU,
            5:o.conversionLitroAtmKWHR,
            6:o.conversionLitroAtmHPHR,
            7:o.conversionLitroAtmPielb,
            }
    numero=int(input("numero? "))
    return switch_dict[opcion](numero)  


main()
